app/src/parser/parser.ts

```diff
@@ -197,39 +200,54 @@ function httpsGet(url: string): Promise<IncomingMessage> {
 
 async function addGroundAltitude(fixes: Fix[]): Promise<unknown> {
   let numErrors = 0;
+  let queuePromises: Promise<void>[] = [];
   for (let i = 0; i < fixes.length; i++) {
-    const fix = fixes[i];
-    const lat = fix.lat;
-    const south = Math.floor(lat);
-    const lon = fix.lon;
-    const west = Math.floor(lon);
-    let buffer: Buffer | null = null;
-    const url = getHgtUrl(south, west);
-    if (!hgtCache.has(url)) {
+    queuePromises.push((async () => {
```

Comment:
For a track with 100k fixes it would be 100k promises and 100k closures.

I think it would be preferable to split the loop in 2:
1) loop over the fixes and download the data
2) loop over the fixes again and extract the elevation.

That would be iterating twice over the fixes but probably as fast as creating and garbage collecting a few hundred thousand objects.